### PR TITLE
Fix WAN notification 

### PR DESCRIPTION
**Pull request by stephdl**
The notifications of wan are not sent when the wan is down 

https://github.com/NethServer/dev/issues/6497

Screenshot - 2021-04-26T191027 986
Screenshot - 2021-04-26T191010 526

### CODE REVIEW

File: api/wan/read [hunk 1]
```diff
@@ -74,7 +74,7 @@ if($cmd eq 'providers') {
     $ret->{'configuration'}{'interfaces'} = \@interfaces;
     my @tmp = split(",",$cdb->get_prop('firewall','CheckIP'));
     $ret->{'configuration'}{'multiwan'}{'CheckIP'} = \@tmp;
-    foreach (qw(MaxNumberPacketLoss MaxPercentPacketLoss NotifyWan PingInterval WanMode)) {
+    foreach (qw(MaxNumberPacketLoss MaxPercentPacketLoss NotifyWan NotifyWanFrom NotifyWanTo PingInterval WanMode)) {
```
Comment (DavidePrincipi) [inline api/wan/read:77]: Question: shall we consider root/EmailAddress and root/SenderAddress as default values for the NotifyWanFrom and NotifyWanTo props?

**Review by DavidePrincipi — COMMENTED** [review]
The root/EmailAddress and root/SenderAddress props are now ignored...

**Comment by stephdl** [comment]
> The root/EmailAddress and root/SenderAddress props are now ignored...

Yes it is a breaking change introduced, actually if you migrate from NethGui to cockpit and you save the form of wan notification, you won't ever receive a notification except if you have given a root email address.

I proposed to give back to the NethGui behaviour but we could fix the cockpit side by disabling the wan notification until the root email is set.

**Review by DavidePrincipi — APPROVED** [review]

File: api/wan/read (hunk 1 above)
Comment (gsanchietti) [reply api/wan/read:77]: Of course!

The NotifyWanFrom NotifyWanTo  should be completely ignored by cockpit since have been deprecated.

**Review by gsanchietti — CHANGES_REQUESTED** [review]
Cockpit has been designed to hide multiple mail addresses for notification.
The user should configure only the root notification page present inside the Settings page.

So please do not add new fields to the UI, just fix the bug at API level by reading the root key from configuration db.

**Comment by DavidePrincipi** [comment]
I checked the Nethgui behavior:

https://github.com/NethServer/nethserver-firewall-base/blob/ef3b3f5f83debbc8f1a54918eaaeffdc9316fb2f/root/usr/share/nethesis/NethServer/Module/Providers.php#L51-L52

There's no initialization based on root/EmailAddress and root/SenderAddress: it's ok to restore the original Nethgui behavior.

**Comment by DavidePrincipi** [comment]
> So please do not add new fields to the UI, just fix the bug at API level by reading the root key from configuration db.

No problem to go in that direction. Shall we change this logic?

https://github.com/NethServer/nethserver-lsm/blob/d0e2a0f2bf89e86dada255941222fde9eaa1bd7a/root/usr/libexec/nethserver/lsm-wan-notify#L37-L38

What I mean is:

- If NotifyWanFrom is not set, use root/SenderAddress
- If NotifyWanTo is not set, use root/EmailAddress

**Comment by gsanchietti** [comment]
> No problem to go in that direction. Shall we change this logic?

It shouldn't be needed, because the values of those properties are already handled by Cockpit API: https://github.com/NethServer/nethserver-firewall-base/blob/ef3b3f5f83debbc8f1a54918eaaeffdc9316fb2f/api/wan/update#L50

I think we need to fix only the logic inside the API itself.

**Comment by stephdl** [comment]
obsoleted by  https://github.com/NethServer/nethserver-firewall-base/pull/186

**Pull request closed without merging**